phoenicis-containers/src/main/java/org/phoenicis/containers/wine/WinePrefixContainersManager.java

```diff
@@ -84,21 +85,19 @@ public void fetchContainers(Consumer<List<ContainerCategoryDTO>> callback, Consu
                 final String winePrefixName = winePrefixPath.substring(winePrefixPath.lastIndexOf('/') + 1);
 
                 // find shortcuts which use this container
-                List<ShortcutDTO> shortcutDTOS = new ArrayList<>();
-                libraryManager.fetchShortcuts().forEach(shortcutCategoryDTO -> {
-                    shortcutCategoryDTO.getShortcuts().forEach(shortcutDTO -> {
-                        try {
-                            final Map<String, Object> shortcutProperties = objectMapper
-                                    .readValue(shortcutDTO.getScript(), new TypeReference<Map<String, Object>>() {
-                                    });
-                            if (shortcutProperties.get("winePrefix").equals(winePrefixName)) {
-                                shortcutDTOS.add(shortcutDTO);
+                List<ShortcutDTO> shortcutDTOS = libraryManager.fetchShortcuts().stream()
+                        .flatMap(shortcutCategory -> shortcutCategory.getShortcuts().stream()).filter(shortcut -> {
+                            boolean toAdd = false;
+                            try {
+                                final Map<String, Object> shortcutProperties = objectMapper
```

Comment:
I thought about it again: we cannot have it because it's engine specific. The only thing we could do is to implement this in the Wine shortcut script in JS.